File: Documentation/5.4/Samples/python/ClientApi/Session/HowTo/Refresh.py
from examples_base import ExampleBase, Employee


class Foo:
    # region refresh_1
    def refresh(self, entity: object) -> object: ...

    # endregion


class HowToRefresh(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_refresh(self):
        with self.embedded_server.get_document_store("Refresh") as store:
            with store.open_session() as session:
                # region refresh_2
                employee = session.load("employees/1", Employee)
                self.assertEquals("Doe", employee.last_name)

                # LastName changed to "Shmoe"

                session.advanced.refresh(employee)
                self.assertEquals("Shmoe", employee.last_name)
                # endregion

File: Documentation/5.4/Samples/php/ClientApi/session/HowTo/Refresh.php
<?php

use PHPUnit\Framework\TestCase;
use RavenDB\Documents\DocumentStore;

interface FooInterface
{
    # region refresh_1
    public function refresh(object $entity): void;
    # endregion
}

class Refresh extends TestCase
{
    public function example(): void
    {
        $store = new DocumentStore();
        try {
            $session = $store->openSession();
            try {
                # region refresh_2
                $employee = $session->load(Employee::class, "employees/1");
                $this->assertEquals("Doe", $employee->getLastName());

                // LastName changed to 'Shmoe'

                $session->advanced()->refresh($employee);

                $this->assertEquals("Shmoe", $employee->getLastName());
                # endregion
            } finally {
                $session->close();
            }
        } finally {
            $store->close();
        }
    }
}
